src/components/SearchMap/SearchMapWithGoogleMaps.js

```diff
@@ -422,29 +418,77 @@ class SearchMapWithGoogleMap extends Component {
       if (!isEqual(this.props.bounds, currentBounds) && !this.viewportBounds) {
         fitMapToBounds(this.map, this.props.bounds, { padding: 0 });
       }
+
+      if (prevProps.infoCardOpen !== this.props.infoCardOpen) {
+        this.map.setOptions({ disableDoubleClickZoom: !!this.props.infoCardOpen });
+      }
+    }
+
+    if (!this.map && this.state.mapContainer) {
+      this.initializeMap();
+
+      /* Notify parent component that the map is loaded */
+      this.props.onMapLoad(this.map);
+    } else if (prevProps.mapComponentRefreshToken !== this.props.mapComponentRefreshToken) {
+      /* Notify parent component that the map is loaded */
+      this.props.onMapLoad(this.map);
     }
   }
 
-  onMapLoad(map) {
-    this.map = map;
-    this.props.onMapLoad(map);
+  componentWillUnmount() {
+    this.idleListener.remove();
+  }
+
+  initializeMap() {
+    const { offsetHeight, offsetWidth } = this.state.mapContainer;
+    const hasDimensions = offsetHeight > 0 && offsetWidth > 0;
+
+    if (hasDimensions) {
+      const { bounds, center, zoom } = this.props;
+      const maps = window.google.maps;
+      const controlPosition = window.google.maps.ControlPosition.LEFT_TOP;
+      const zoomAndCenter =
+        !bounds && !center ? { zoom: 1, center: { lat: 0, lng: 0 } } : { zoom, center };
```

Comment:
Well that's a one-liner! I think this would be more clear the other way. You check if `zoom` and `center` are defined and use them, or you provide a default object:

```suggestion
        bounds && center
          ? { zoom, center }
          : { zoom: 1, center: { lat: 0, lng: 0 } };
```

However, that is not equivalent to the code in the PR. Is the logic correct? Do you really want to use `{ zoom, center }` if either one is defined?

Reply:
When at least one of them (bounds or center) is defined, we want to use them:
- If bounds are defined => map should be fitted to bounds.
- If only the center is defined => map should show that place (i.e. origin) in given or default zoom-level (e.g. street address: Erottajankatu 19 )

Only in this specific case where no bounds or center exists, we default to the center of the world/Mercator projection and show it with most-zoomed out viewport => to show the whole earth.

Prettier gave this formatting, but since this seems to be hard to understand one-liner, I'll format it:
```js
const zoomOutToShowEarth = { zoom: 1, center: { lat: 0, lng: 0 } };
const zoomAndCenter = !bounds && !center ? zoomOutToShowEarth : { zoom, center };
```